src/Misc/environmentTextureTools.ts

```diff
@@ -715,7 +715,7 @@ export class EnvironmentTextureTools {
     }
 
     /** @hidden */
-    public static _UpdateRGBDAsync(internalTexture: InternalTexture, data: ArrayBufferView[][], sphericalPolynomial: Nullable<SphericalPolynomial>, lodScale: number, lodOffset: number): Promise<void> {
+    public static _UpdateRGBDAsync(internalTexture: InternalTexture, data: ArrayBufferView[][], sphericalPolynomial: Nullable<SphericalPolynomial>, lodScale: number, lodOffset: number): Promise<InternalTexture> {
         internalTexture._source = InternalTextureSource.CubeRawRGBD;

```

Comment:
Should this not add the callback as well for now ?

Reply:
well, this is the callback. It shouldn't register itself.

This PR is meant to decouple the rebuild from internal texture, this is just one specific case which I tried not to change too much.

Reply:
Whereas I agree that the current architecture doesn't make too much sense, I don't see where rebuild is being triggered